packages/langium/src/utils/cst-util.ts

```diff
@@ -74,6 +74,36 @@ export function toDocumentSegment(node?: CstNode): DocumentSegment | undefined {
     };
 }
 
+export enum RangeComparison {
+    Before = 0,
+    OverlapFront = 1,
+    Inside = 2,
+    OverlapBack = 3,
+    After = 4
```

Comment:
If you order these differently, you could use `<` or `>` comparisons to check whether there's an overlap.